code/game/objects/items/melee/misc.dm

```diff
@@ -315,3 +315,110 @@
 	..()
+
+/obj/item/melee/roastingstick
+	name = "advanced roasting stick"
+	desc = "A telescopic roasting stick with a miniature shield generator designed to ensure entry into various high-tech shielded cooking ovens and firepits."
+	icon_state = "roastingstick_0"
+	item_state = "null"
+	slot_flags = SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	item_flags = NONE
+	force = 0
+	attack_verb = list("hit", "poked")
+	var/obj/item/reagent_containers/food/snacks/sausage/held_sausage
+	var/static/list/ovens
+	var/on = FALSE
+
+/obj/item/melee/roastingstick/Initialize()
+	. = ..()
+	if (!ovens)
+		ovens = typecacheof(list(/obj/singularity, /obj/machinery/power/supermatter_shard/crystal, /obj/structure/bonfire))
+
+/obj/item/melee/roastingstick/attack_self(mob/user)
+	on = !on
+	if(on)
+		extend(user)
+	else
+		if (held_sausage)
+			to_chat(user, "<span class='warning'>You can't retract [src] while [held_sausage] is attached!</span>")
+			return
+		retract(user)
+
+	playsound(src.loc, 'sound/weapons/batonextend.ogg', 50, 1)
+	add_fingerprint(user)
+
+/obj/item/melee/roastingstick/attackby(atom/target, mob/user)
+	..()
+	if (istype(target, /obj/item/reagent_containers/food/snacks/sausage))
+		if (!on)
+			to_chat(user, "<span class='warning'>You must extend [src] to attach anything to it!</span>")
+			return
+		if (held_sausage)
+			to_chat(user, "<span class='warning'>[held_sausage] is already attached to [src]!</span>")
+			return
+		if (user.transferItemToLoc(target, src))
+			held_sausage = target
+		else
+			to_chat(user, "<span class='warning'>[target] doesn't seem to want to get on [src]!</span>")
+	update_icon()
+
+/obj/item/melee/roastingstick/attack_hand(mob/user)
+	..()
+	if (held_sausage)
+		user.put_in_hands(held_sausage)
+		held_sausage = null
+	update_icon()
+
+/obj/item/melee/roastingstick/update_icon()
+	. = ..()
+	cut_overlays()
+	if (held_sausage)
+		var/mutable_appearance/sausage = mutable_appearance(icon, "roastingstick_sausage")
+		add_overlay(sausage)
+
+/obj/item/melee/roastingstick/proc/extend(user)
+	to_chat(user, "<span class ='warning'>You extend [src].</span>")
+	icon_state = "roastingstick_1"
+	item_state = "nullrod"
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/melee/roastingstick/proc/retract(user)
+	to_chat(user, "<span class ='notice'>You collapse [src].</span>")
+	icon_state = "roastingstick_0"
+	item_state = null
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/melee/roastingstick/handle_atom_del(atom/target)
+	if (target == held_sausage)
+		held_sausage = null
+		update_icon()
+
+/obj/item/melee/roastingstick/afterattack(atom/target, mob/user, proximity)
+	if (!on)
+		return
+	if (is_type_in_typecache(target, ovens))
+		if (held_sausage && held_sausage.roasted)
+			to_chat("Your [held_sausage] has already been cooked.")
+			return
+		if (istype(target, /obj/singularity) && get_dist(user, target) < 10)
+			to_chat(user, "You send [held_sausage] towards [target].")
+			playsound(src, 'sound/items/rped.ogg', 50, 1)
```

Comment:
just to be sure, this is the bluespace rped sound and not the "replacing parts" noise from the regular rped?

Reply:
https://github.com/tgstation/tgstation/blob/4211482e4be06f5e59185776f98ecca6e6b00371/code/modules/research/stock_parts.dm#L21

I guess it could have the beam sound too, i just wanted a more advanced deployment than the extendostick.